modules/core/shared/src/main/scala/lucuma/core/model/Tracking.scala

```diff
@@ -80,3 +95,38 @@ trait EphemerisCoordinatesOptics {
     delta.andThen(Offset.q)
 
 }
+
+case class SiderealCoordinates(override val coord: Coordinates) extends TrackedCoordinates
+
+object SiderealCoordinates extends SiderealCoordinatesOptics {
+  given Eq[SiderealCoordinates] = Eq.by(_.coord)
+}
+
+trait SiderealCoordinatesOptics {
+  /** @group Optics */
+  val coordinates: Lens[SiderealCoordinates, Coordinates] =
+    Focus[SiderealCoordinates](_.coord)
+
+  /** @group Optics */
+  val rightAscension: Lens[SiderealCoordinates, RightAscension] =
+    coordinates.andThen(Coordinates.rightAscension)
+
+  /** @group Optics */
+  val declination: Lens[SiderealCoordinates, Declination] =
+    coordinates.andThen(Coordinates.declination)
+}
+
+/**
+ * Implementors can calculate the coordinates for a given object at a certain timestamp
+ */
+trait Tracking {
+  def at(i: Instant): Option[TrackedCoordinates]
+  def atEpoch: TrackedCoordinates
```

Comment:
I added this method to give use the coordinates at epoch. For sidereals is fairly straightforward but for non-sidereals I'd use the convention that epoch would be the first item on the ephemeris.
Note this implies that we need to have at least one entry on the ephemeris table
I discussed this with Andy and he agrees that we should not have non-sidereal targets without ephemeris but you may disagree

An alternative is to make the signature
```
def atEpoch: Option[TrackedCoordinates]
```
But it does complicate a bit the code downstream where it is simpler if we can avoid the `Option`
WDYT?